src/lib/ext-activitylog.js

```diff
@@ -156,28 +156,32 @@ class View {
   }
 
   handleEvent(event) {
-    if (event.type === 'click') {
-      switch (event.target) {
-        case this.clearLogBtn:
-          this.clearLogBtn.dispatchEvent(new CustomEvent('clearlog'));
-          break;
-        case this.optionsBtn:
-          dropDownController.toggleDropDown(this.optionsBtn);
-          break;
-        case this.saveLogBtn:
-          this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
-          break;
+    try {
+      if (event.type === 'click') {
+        switch (event.target) {
+          case this.clearLogBtn:
+            this.clearLogBtn.dispatchEvent(new CustomEvent('clearlog'));
+            break;
+          case this.optionsBtn:
+            dropDownController.toggleDropDown(this.optionsBtn);
+            break;
+          case this.saveLogBtn:
+            this.saveLogBtn.dispatchEvent(new CustomEvent('savelog'));
+            break;
+        }
+      } else if (event.type === 'change' && event.target === this.loadLogFile) {
+        const logFile = event.target.files[0];
+        this.loadLogFile.value = '';
+        this.loadLogFile.dispatchEvent(
+          new CustomEvent('loadlog', { detail: logFile })
+        );
+      } else if (event.type === 'logcountchange') {
+        this.updateLogCounter(event.detail);
+      } else {
+        throw new Error(`wrong event type - ${event.type}`);
       }
-    } else if (event.type === 'change' && event.target === this.loadLogFile) {
-      const logFile = event.target.files[0];
-      this.loadLogFile.value = '';
-      this.loadLogFile.dispatchEvent(
-        new CustomEvent('loadlog', { detail: logFile })
-      );
-    } else if (event.type === 'logcountchange') {
-      this.updateLogCounter(event.detail);
-    } else {
-      throw new Error(`wrong event type - ${event.type}`);
+    } catch (error) {
+      this.setError(error.message);
```

Comment:
would you mind to describe me the reasons why we are wrapping all the handleEvent method (this one and also the one defined at line 351) in a try...catch? 
is it a legit scenario or just something happening in the test case? 

(I'm basically wondering if it is the test case that should provide a more realistic testing environment, otherwise we be handling errors at every "layer", https://en.wikipedia.org/wiki/Defensive_programming may be an interesting read on the topic)

Reply:
I just rechecked that part of the code. There should no error thrown in the try catch block at line 183. So, I have removed this try and catch block.

At line 351, there is no need of try catch block since I have been handling the errors inside every methods (which have been called from `handleEvent`) separately. I made a helper function to catch the error from `sendMessage` API at a single place. The helper function communicates with the backend.
```javascript
// helper function
async sendMsgToBackground({ requestType, requestParams }) {
  try {
    return await browser.runtime.sendMessage({
      requestTo: 'ext-monitor',
      requestType,
      requestParams,
    });
  } catch (error) {
    this.view.setError(error.message);
  }
}

```
You can view the changes in this commit - 6e0583203